src/core/interactive/ViewerPreferences.ts

```diff
@@ -18,20 +45,353 @@ class ViewerPreferences {
     this.dict = dict;
   }
 
-  DisplayDocTitle(): PDFBool | undefined {
-    const DisplayDocTitle = this.dict.lookup(PDFName.of('DisplayDocTitle'));
-    if (DisplayDocTitle instanceof PDFBool) return DisplayDocTitle;
+  protected FindBool(key: BoolViewerPrefKey): PDFBool | undefined {
+    const returnObj = this.dict.lookup(PDFName.of(key));
+    if (returnObj instanceof PDFBool) return returnObj;
+    return undefined;
+  }
+
+  protected FindNameVal(key: NameViewerPrefKey): PDFName | undefined {
+    const returnObj = this.dict.lookup(PDFName.of(key));
+    if (returnObj instanceof PDFName) return returnObj;
     return undefined;
   }
 
+  protected FindArray(key: ArrayViewerPrefKey): PDFArray | undefined {
+    const returnObj = this.dict.lookup(PDFName.of(key));
+    if (returnObj instanceof PDFArray) return returnObj;
+    return undefined;
+  }
+
+  protected FindNumber(key: NumberViewerPrefKey): PDFNumber | undefined {
+    const returnObj = this.dict.lookup(PDFName.of(key));
+    if (returnObj instanceof PDFNumber) return returnObj;
+    return undefined;
+  }
+
+  /*
+   * Gets the value of the flag specifying whether to hide the conforming reader’s tool bars when the document is active.
+   * Default value if not specified: false.
+   */
```

Comment:
These comments are good, but please tweak them a bit to confirm to the comment structure found in other parts of the pdf-lib codebase. Comment structure consistency is as important as code consistency in pdf-lib (even more so for end-users). 

Here's an example of the comment structure for boolean methods: https://github.com/Hopding/pdf-lib/blob/4be96eb269910bbc6d3ddcd48f2a6382742e179b/src/api/form/PDFDropdown.ts#L266-L279

This comment can be rewritten as follows to adhere to this structure:

```suggestion
  /**
   * Returns `true` if PDF readers should hide toolbar menus when displaying
   * this document.
   * @returns Whether or not toolbar menus should be hidden.
   */
```

The code example can be omitted from the comment since this method is part of the `src/core` module rather than the `src/api` module. Though it wouldn't hurt to include anyways. But it's not required.

It would be time consuming to explain in detail exactly what the comment structure is. But I trust the codebase provides sufficient examples to make it clear. The main thing is to keep the wording, grammar, flow, and voice of the comments similar. However, some specific things to watch out for are:
* Use doc-comments `/** ... */` (as opposed to `/* ... */`), unless the comment is a developer note.
* Comments should not exceed the 80 character line limit. Unfortunately the linter and formatter don't check this, so you'll need to do it yourself (VSCode can display a vertical marker at the 80 character point that makes this easy).
* Return types should be documented with an `@returns` tag and an appropriate description.
* Parameter types should be documented with an `@param` tag and an appropriate description.

This note applies to several of the following methods as well, though I won't leave comments on all of them.

Reply:
I have attempted to alter the comments in the code, as per this conversation. The use of string `enums` does not seem to be used elsewhere in the library, so I have no great proforma. there is still an inconsistency with the possible values being listed in the description for the setters and in the `@returns` for the getters. Please let me know which style you prefer.

Reply:
@mcshaz the new comments (and all your other changes) look great! Thanks for addressing my review comments so promptly.

The preferred way to document enum values is via doc comments on the enum value definitions themselves. For example:
```ts
export enum NonFullScreenPageMode {
  /** 
   * After exiting full-screen mode neither the document outline nor thumbnail 
   * images should be visible. 
   */
  UseNone = 'UseNone',

  /** After exiting full-screen mode the document outline should be visible. */
  UseOutlines = 'UseOutlines',

  /** After exiting full-screen mode the thumbnail images should be visible. */
  UseThumbs = 'UseThumbs',

  /** 
   * After exiting full-screen mode the optional content group panel should be 
   * visible. 
   */
  UseOC = 'UseOC',
}
```

This removes the need for duplicate information in the comments, especially if the enum is used in multiple methods. The comments will display as follows in the API docs: 

![image](https://user-images.githubusercontent.com/10351828/104930854-48a3d000-596b-11eb-95b0-acc603a57eb6.png)

Reply:
Thank you - I have attempted to address all the suggestions. I now also note that table 28 on page 73 of the spec includes entries in the `catalog` dictionary `PageLayout` and `PageMode`. I cannot see why these are not `ViewerPreferences` in their specification, but c'est la vie. I can add the associated getters/setters for these to the PDFCatalog class in the same manner if you would like - simply because it kind of rounds off the ViewerPreferences, and in truth `NonFullScreenPageMode` does not make much sense without  at least PageMode. Note the documentation now has the example:
```js
// set the PageMode, otherwise the subsequent NonFullScreenPageMode has no meaning
pdfDoc.catalog.set(PDFName.of('PageMode'),PDFName.of('FullScreen'))
// set what happens when fullScreen is closed
viewerPrefs.setNonFullScreenPageMode(NonFullScreenPageMode.UseOutlines) 
```

Reply:
I'd be happy to accept a PR for adding those setters/getters to the `PDFCatalog` class 👍